pkg/services/ngalert/models/alert_rule.go

```diff
@@ -98,6 +98,13 @@ const (
 
 	// FolderTitleLabel is the label that will contain the title of an alert's folder/namespace.
 	FolderTitleLabel = GrafanaReservedLabelPrefix + "folder"
+
+	// StateReasonAnnotation is annotation that will contain a reason for the state in the evaluation state is different from the alert state (for example, mapped NoData or Error state or StateReasonMissingSeries).
```

Comment:
This sentence is a little difficult to understand, perhaps something like:

```suggestion
	// StateReasonAnnotation is the name of the annotation that explains the difference between evaluation state and alert state (i.e. changing state when NoData or Error).
```